homeassistant/components/nederlandse_spoorwegen/sensor.py

```diff
@@ -47,13 +47,22 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the departure sensor."""
 
     nsapi = ns_api.NSAPI(config[CONF_API_KEY])
+
     try:
         stations = nsapi.get_stations()
     except (
         requests.exceptions.ConnectionError,
         requests.exceptions.HTTPError,
     ) as error:
-        _LOGGER.error("Couldn't fetch stations, API key correct?: %s", error)
+        _LOGGER.error("NS integration could not connect to the internet: %s", error)
+        return
+    except Exception as error:
+        _LOGGER.error("NS integration crashed, please check configuration: %s", error)
```

Comment:
What about the configuration might be wrong in this case?

Reply:
A bad API key (NS has quite the complicated API key management system) or bad station codes come to mind. NSAPI accepts everything and (for now) always assumes a trip is given back. If the NS replies saying something in the config is not correct (key, station codes), NSAPI crashes and this is not captured by the current integration implementation causing an error log without meaning ("payload is missing").

Reply:
What error is raised by the library?

Reply:
```
Traceback (most recent call last):
  File "/home/yarmo/gh/home-assistant/homeassistant/helpers/entity_platform.py", line 179, in _async_setup_platform
    await asyncio.wait_for(asyncio.shield(task), SLOW_SETUP_MAX_WAIT)
  File "/usr/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
    return fut.result()
  File "/usr/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/yarmo/gh/home-assistant/homeassistant/components/nederlandse_spoorwegen/sensor.py", line 51, in setup_platform
    stations = nsapi.get_stations()
  File "/home/yarmo/gh/home-assistant/venv/lib/python3.7/site-packages/ns_api.py", line 951, in get_stations
    return self.parse_stations(raw_stations)
  File "/home/yarmo/gh/home-assistant/venv/lib/python3.7/site-packages/ns_api.py", line 939, in parse_stations
    for station in obj['payload']:
KeyError: 'payload'
```

Reply:
The library should handle the missing payload and inform the library user, eg by raising a library specific error.

Please update the library first.

Reply:
Will do, thanks for your review!